sockjsclient/xhr.go

```diff
@@ -92,61 +86,84 @@ func (x *XHRSession) Recv() (string, error) {
 
 	// start to poll from the server until we receive something
 	for {
-		resp, err := x.client.Post(x.sessionURL+"/xhr", "text/plain", nil)
+		req, err := http.NewRequest("POST", x.sessionURL+"/xhr", nil)
 		if err != nil {
-			return "", err
+			return "", fmt.Errorf("Receiving data failed: %s", err)
 		}
-		defer resp.Body.Close()
 
-		if resp.StatusCode != http.StatusOK {
-			return "", fmt.Errorf("Receiving data failed. Want: %d Got: %d",
-				http.StatusOK, resp.StatusCode)
-		}
+		req.Header.Set("Content-Type", "text/plain")
+
+		select {
+		case <-x.abort:
```

Comment:
i was going to suggest to use ctxhttp but it seems they removed go1.4 support from their upstream